src/Features/LanguageServer/Protocol/Handler/Completion/CompletionResolveHandler.cs

```diff
@@ -185,5 +188,40 @@ private static bool MatchesLSPCompletionItem(LSP.CompletionItem lspCompletionIte
 
             return cacheEntry;
         }
+
+        // Certain language servers such as Razor may want TextEdits formatted using their own options instead of C#/VB options.
+        private static async Task<OptionSet> GetCompletionFormattingOptionsAsync(Document document, CancellationToken cancellationToken)
+        {
+            var documentOptions = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+
+            var optionsService = document.Services.GetService<IDocumentOptionsProvider>();
+            if (optionsService == null)
+            {
+                return documentOptions;
+            }
+
+            var options = await optionsService.GetOptionsForDocumentAsync(document, cancellationToken).ConfigureAwait(false);
+            if (options == null)
+            {
+                return documentOptions;
+            }
+
+            if (options.TryGetDocumentOption(new OptionKey(FormattingOptions.UseTabs, document.Project.Language), out var useTabs) &&
+                useTabs != null &&
+                useTabs.GetType() == typeof(bool))
+            {
+                documentOptions = documentOptions.WithChangedOption(FormattingOptions.UseTabs, (bool)useTabs);
+            }
+
+            if (options.TryGetDocumentOption(new OptionKey(FormattingOptions.TabSize, document.Project.Language), out var tabSize) &&
+                tabSize != null &&
+                tabSize.GetType() == typeof(int))
+            {
+                documentOptions = documentOptions.WithChangedOption(FormattingOptions.TabSize, (int)tabSize);
+                documentOptions = documentOptions.WithChangedOption(FormattingOptions.IndentationSize, (int)tabSize);
+            }
```

Comment:
Seems odd that these three specific options are hard-coded here. Should this be moved to the Razor-specific code, as that is the thing that knows that Razor cares about these three options.